examples/hitl/rearrange_v2/world.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Dict, List, Optional, Set
+
+from habitat.sims.habitat_simulator import sim_utilities
+from habitat.tasks.rearrange.articulated_agent_manager import (
+    ArticulatedAgentManager,
+)
+from habitat.tasks.rearrange.rearrange_sim import RearrangeSim
+from habitat_sim.physics import ManagedArticulatedObject
+
+
+class World:
+    """
+    Global world state shared by each user.
+    """
+
+    def __init__(
+        self,
+        sim: RearrangeSim,
+    ):
+        self._sim = sim
+
+        # Cache of all held objects.
+        # Prevents users from picking objects held by others.
+        self._all_held_object_ids: Set[int] = set()
+        # Cache of all opened articulated object links.
+        self._opened_link_set: Set = set()
```

Comment:
As discussed, I'll be changing the code to query open/close state of objects rather than caching them here.

This is an old approach from before utilities were merged in.